docs(overhaul): provide a very detailed documentation for the options (#456)
actions/stale: README.md

| [exempt-assignees](#exempt-assignees)                               | Assignees on issues/PRs exempted from stale              |                       |
| [exempt-issue-assignees](#exempt-issue-assignees)                   | Override [exempt-assignees](#exempt-assignees)           |                       |
| [exempt-pr-assignees](#exempt-pr-assignees)                         | Override [exempt-assignees](#exempt-assignees)           |                       |
| [exempt-all-assignees](#exempt-all-assignees)                       | Exempt all issues/PRs with assignees from stale          |                       |
| [exempt-all-issue-assignees](#exempt-all-issue-assignees)           | Override [exempt-all-assignees](#exempt-all-assignees)   |                       |
| [exempt-all-pr-assignees](#exempt-all-pr-assignees)                 | Override [exempt-all-assignees](#exempt-all-assignees)   |                       |
| [enable-statistics](#enable-statistics)                             | Display statistics in the logs                           | `true`                |

### Detailed options

#### repo-token

> PAT (Personal Access Token) to allow the stale workflow to authenticate and perform API calls to GitHub.  
> Under the hook, used by [@actions/github](https://www.npmjs.com/package/@actions/github).

Reviewer: ```suggestion
Personal Access Token (PAT) that allows the stale workflow to authenticate and perform API calls to GitHub.  
Under the hood, it uses the [@actions/github](https://www.npmjs.com/package/@actions/github) package.
```

Default value: `${{ github.token }}`

#### days-before-stale

The idle number of days before marking the issues or the pull requests as stale (by adding a label).  
The issues or the pull requests will be marked as stale if the last update (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) is older than the idle number of days.

> If set to a negative number like `-1`, the issues or the pull requests will never be marked as stale automatically.  

Reviewer: ```suggestion
If set to a negative number like `-1`, no issues or pull requests will be marked as stale automatically.  
```

Default value: unset

#### days-before-close

The idle number of days before closing the stale issues or the stale pull requests (due to the stale label).  
The issues or the pull requests will be closed if the last update (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) is older than the idle number of days.  
Since adding the stale label will alter the last update date, we can calculate the number of days from this date.

If set to a negative number like `-1`, the issues or the pull requests will never be closed automatically.

The label used to stale is defined by these two options:

- [stale-issue-label](#stale-issue-label)
- [stale-pr-label](#stale-pr-label)

Default value: `7`

#### days-before-issue-close

Override [days-before-close](#days-before-close) but only for the idle number of days before closing the stale issues.

Default value: unset

#### days-before-pr-close

Override [days-before-close](#days-before-close) but only for the idle number of days before closing the stale pull requests.

Default value: unset

#### stale-issue-message

> The message that will be added as a comment to the issues when the stale workflow mark it automatically as stale with a label.

Reviewer: ```suggestion
The message that will be added as a comment to the issues when the stale workflow marks it automatically as stale with a label.
```

You can omit the comment by setting [skip-stale-issue-message](#skip-stale-issue-message) to `true`.

Default value: unset

#### stale-pr-message

> The message that will be added as a comment to the pull requests when the stale workflow mark it automatically as stale with a label.

Reviewer: ```suggestion
The message that will be added as a comment to the pull requests when the stale workflow marks it automatically as stale with a label.
```

You can omit the comment by setting [skip-stale-pr-message](#skip-stale-pr-message) to `true`.

Default value: unset

#### close-issue-message

> The message that will be added as a comment to the issues when the stale workflow close it automatically after being stale for too long.

Reviewer: ```suggestion
The message that will be added as a comment to the issues when the stale workflow closes it automatically after being stale for too long.
```

Default value: unset

#### close-pr-message

> The message that will be added as a comment to the pull requests when the stale workflow close it automatically after being stale for too long.

Reviewer: ```suggestion
The message that will be added as a comment to the pull requests when the stale workflow closes it automatically after being stale for too long.
```

Default value: unset

#### stale-issue-label

The label that will be added to the issues when automatically marked as stale.  

> It should also be the one you should use if you wish to manually mark as stale the issues.

Reviewer: ```suggestion
Use this label when you want to manually mark a issue as stale.
```

Author: ```suggestion
If you wish to speedup the stale workflow for the issues, you can add this label manually to mark as stale.
```

Default value: `Stale`

#### close-issue-label

The label that will be added to the issues when closed automatically.  
It will be automatically removed if the issues are no longer closed nor locked.

Default value: unset

#### stale-pr-label

The label that will be added to the pull requests when automatically marked as stale.  

> It should also be the one you should use if you wish to manually mark as stale the pull requests.

Reviewer: ```suggestion
Use this label when you want to manually mark a pull request as stale.
```

Author: ```suggestion
If you wish to speedup the stale workflow for the pull requests, you can add this label manually to mark as stale.
```

Default value: `Stale`

#### close-pr-label

The label that will be added to the pull requests when closed automatically.  
It will be automatically removed if the pull requests are no longer closed nor locked.

Default value: unset

#### exempt-issue-labels

The label(s) that can exempt to automatically mark as stale the issues.  
It can be a comma separated list of labels (e.g: `question,bug`).

If unset (or an empty string), this option will not alter the stale workflow.

Default value: unset

#### exempt-pr-labels

The label(s) that can exempt to automatically mark as stale the pull requests.  
It can be a comma separated list of labels (e.g: `need-help,WIP`).

If unset (or an empty string), this option will not alter the stale workflow.

Default value: unset

#### only-labels

> A white-list of label(s) to only process the issues or the pull requests that contain all these label(s).  

Reviewer: ```suggestion
A allow-list of label(s) to only process the issues or the pull requests that contain all these label(s).  
```

I think the trend is to use allow-list and block-list instead of white-list and black-list.

Author: ```suggestion
An allow-list of label(s) to only process the issues or the pull requests that contain all these label(s).  
```

It can be a comma separated list of labels (e.g: `answered,needs-rebase`).

If unset (or an empty string), this option will not alter the stale workflow.

If you wish to only check that the issues or the pull requests contain one of these label(s), use instead [any-of-labels](#any-of-labels).

Default value: unset

#### only-issue-labels

Override [only-labels](#only-labels) but only to process the issues that contain all these label(s).

Default value: unset

#### only-pr-labels

Override [only-labels](#only-labels) but only to process the pull requests that contain all these label(s).

Default value: unset

#### any-of-labels

> A white-list of label(s) to only process the issues or the pull requests that contain one of these label(s).  

Reviewer: ```suggestion
A allow-list of label(s) to only process the issues or the pull requests that contain one of these label(s).  
```

Author: ```suggestion
An allow-list of label(s) to only process the issues or the pull requests that contain one of these label(s).  
```

| [debug-only](#debug-only)                                           | Dry-run                                                  | `false`               |
| [ascending](#ascending)                                             | Order to get issues/PRs                                  | `false`               |
| [skip-stale-issue-message](#skip-stale-issue-message)               | Skip adding stale message on stale issues                | `false`               |
| [skip-stale-pr-message](#skip-stale-pr-message)                     | Skip adding stale message on stale PRs                   | `false`               |
| [start-date](#start-date)                                           | Skip stale action for issues/PRs created before it       |                       |
| [delete-branch](#delete-branch)                                     | Delete branch after closing a stale PR                   | `false`               |
| [exempt-milestones](#exempt-milestones)                             | Milestones on issues/PRs exempted from stale             |                       |

> | [exempt-issue-milestones](#exempt-issue-milestones)                 | Override [exempt-milestones](#exempt-milestones)         |                       |

Reviewer 2: would it make sense for these overrides to be slightly more explicit in the docs like `Override exempt-milestones for issues only`? It might be good to be explicit about what is happening and detail what `override` means in this context

Author: I will make the change tonight

Author: @luketomlinson done 46352757a0418ff8ddd8cd41d21323ed28add94a

Default value: `${{ github.token }}`

#### days-before-stale

The idle number of days before marking the issues or the pull requests as stale (by adding a label).  
The issues or the pull requests will be marked as stale if the last update (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) is older than the idle number of days.

If set to a negative number like `-1`, no issues or pull requests will be marked as stale automatically.  
In that case, you can still add the stale label manually to mark as stale.

The label used to stale is defined by these two options:

- [stale-issue-label](#stale-issue-label)
- [stale-pr-label](#stale-pr-label)

A comment can also be added to notify about the stale and is defined by these two options:

- [stale-issue-message](#stale-issue-message)
- [stale-pr-message](#stale-pr-message)

You can fine tune which issues or pull requests should be marked as stale based on the milestones, the assignees, the creation date and the missing/present labels from these options:

- [exempt-issue-labels](#exempt-issue-labels)
- [exempt-pr-labels](#exempt-pr-labels)

> - [only-labels](#only-labels)

Reviewer 2: dup list item?

Author: a3ba9a6ab3bf585fe7be6510c22b9e7cef490c6a

Default value: `30`

#### remove-stale-when-updated

Automatically remove the stale label when the issues or the pull requests are updated (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) or commented.

Default value: `true`

#### remove-issue-stale-when-updated

Override [remove-stale-when-updated](#remove-stale-when-updated) but only to automatically remove the stale label when the issues are updated (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) or commented.

Default value: unset

#### remove-pr-stale-when-updated

Override [remove-stale-when-updated](#remove-stale-when-updated) but only to automatically remove the stale label when the pull requests are updated (based on [GitHub issue](https://docs.github.com/en/rest/reference/issues) field `updated_at`) or commented.

Default value: unset

#### debug-only

Run the stale workflow as dry-run.  
No GitHub API calls that can alter your issues and pull requests will happen.  
Useful to debug or when you want to configure the stale workflow safely.

Default value: `false`

#### ascending

> Change the order used to fetch the issues and pull requests from GitHub.
> `true` is for ascending.  
> `false` is for descending.

Reviewer: ```suggestion
Change the order used to fetch the issues and pull requests from GitHub:

- `true` is for ascending.
- `false` is for descending.
```

> It can be useful if your repository is processing so much issues and pull requests that you reach the [operations-per-run](#operations-per-run) limit.  

Reviewer: ```suggestion
It can be useful if your repository is processing so many issues and pull requests that you reach the [operations-per-run](#operations-per-run) limit.
```

Author: ```suggestion
It can be useful if your repository is processing so many issues and pull requests that you reach the [operations-per-run](#operations-per-run) limit.  
```

Based on the order, you could prefer to focus on the new content or on the old content of your repository.

Default value: `false`

#### skip-stale-issue-message

If set to `true`, no comment will be added to the issues when they are automatically marked as stale.

If set to `false`, you can define the comment with the [stale-issue-message](#stale-issue-message) option.

Default value: `false`

#### skip-stale-pr-message

If set to `true`, no comment will be added to the pull requests when they are automatically marked as stale.

If set to `false`, you can define the comment with the [stale-pr-message](#stale-pr-message) option.

Default value: `false`

#### start-date

The start date is used to ignore the issues and pull requests created before the start date.  
Particularly useful when you wish to add this stale workflow on an existing repository and only wish to stale the new issues and pull requests.

> If set, the value must be formatted as `ISO 8601` or as `RFC 2822`.

Reviewer: ```suggestion
If set, the dates must be formatted following the `ISO 8601` or `RFC 2822` standard.
```

Author: ```suggestion
If set, the date must be formatted following the `ISO 8601` or `RFC 2822` standard.
```

It can be a comma separated list of milestones (e.g: `V1,next`).

If unset (or an empty string), this option will not alter the stale workflow.

Default value: unset

#### exempt-issue-milestones

Override [exempt-milestones](#exempt-milestones) but only to process the issues that does not contain one of these milestone(s).

Default value: unset

#### exempt-pr-milestones

Override [exempt-milestones](#exempt-milestones) but only to process the pull requests that does not contain one of these milestone(s).

Default value: unset

#### exempt-all-milestones

If set to `true`, the issues or the pull requests with a milestone will not be marked as stale automatically.

Priority over [exempt-milestones](#exempt-milestones).

Default value: `false`

#### exempt-all-issue-milestones

Override [exempt-all-milestones](#exempt-all-milestones) but only to exempt the issues with a milestone to be marked as stale automatically.

Default value: unset

#### exempt-all-pr-milestones

Override [exempt-all-milestones](#exempt-all-milestones) but only to exempt the pull requests with a milestone to be marked as stale automatically.

Default value: unset

#### exempt-assignees

> A white-list of assignee(s) to only process the issues or the pull requests that does not contain one of these assignee(s).  

Reviewer: ```suggestion
A allow-list of assignee(s) to only process the issues or the pull requests that does not contain one of these assignee(s).  
```

Author: ```suggestion
An allow-list of assignee(s) to only process the issues or the pull requests that does not contain one of these assignee(s).  
```

It can be a comma separated list of assignees (e.g: `marco,polo`).

If unset (or an empty string), this option will not alter the stale workflow.

Default value: unset

#### exempt-issue-assignees

Override [exempt-assignees](#exempt-assignees) but only to process the issues that does not contain one of these assignee(s).

Default value: unset

#### exempt-pr-assignees

Override [exempt-assignees](#exempt-assignees) but only to process the pull requests that does not contain one of these assignee(s).

Default value: unset

#### exempt-all-assignees

If set to `true`, the issues or the pull requests with an assignee will not be marked as stale automatically.

Priority over [exempt-assignees](#exempt-assignees).

Default value: `false`

#### exempt-all-issue-assignees

Override [exempt-all-assignees](#exempt-all-assignees) but only to exempt the issues with an assignee to be marked as stale automatically.

Default value: unset

#### exempt-all-pr-assignees

Override [exempt-all-assignees](#exempt-all-assignees) but only to exempt the pull requests with an assignee to be marked as stale automatically.

Default value: unset

#### enable-statistics

> Collect and display some statistics at the end of the stale workflow logs to know what happened during the run.  

Reviewer: ```suggestion
Collects and display statistics at the end of the stale workflow logs to get a summary of what happened during the run.  
```

**Job frequency:**  
You could change the cron job frequency in the stale workflow to run the stale workflow more often.  

> Usually, this is not very helpful, though.

Reviewer: ```suggestion
Usually, this is not very helpful though.
```

I think we can get rid of the second comma.

### Contributing

> Do you wish to contribute?  
> Have a look at the [contributing](CONTRIBUTING.md) file before helping us.

Reviewer: ```suggestion
We welcome contributions!
Please read the [contributing](CONTRIBUTING.md) file before starting your work.
```
